File: ride-service/src/main/java/com/example/ride_service/model/event/DriverAssignedEvent.java
package com.example.ride_service.model.event;

import lombok.Builder;

import java.util.UUID;

@Builder
public record DriverAssignedEvent(
        UUID rideId,
        UUID driverId,
        String driverName,
        UUID carId,
        String carModel,
        String carColor,
        String carBrand,
        String carLicensePlate,
        Integer seats
) {
}

File: driver-service/src/main/kotlin/com/example/driver_service/model/event/DriverAssignedEvent.kt
package com.example.driver_service.model.event

import java.util.UUID

data class DriverAssignedEvent(
    val rideId: UUID,
    val driverId: UUID,
    val driverName: String,
    val carId: UUID,
    val carModel: String,
    val carColor: String,
    val carBrand: String,
    val carLicensePlate: String,
    val seats: Int
)
